room/compiler-processing/src/main/java/androidx/room/compiler/processing/javac/JavacProcessingEnv.kt

```diff
@@ -183,6 +185,46 @@ internal class JavacProcessingEnv(
         } as T
     }
 
+    // TODO: Add a cache layer
+    fun wrapExecutableElement(element: ExecutableElement): JavacExecutableElement {
+        val enclosingType = element.requireEnclosingType(this)
+
+        return when (element.kind) {
+            ElementKind.CONSTRUCTOR -> {
+                JavacConstructorElement(
+                    env = this,
+                    containing = enclosingType,
+                    element = element
+                )
+            }
+            ElementKind.METHOD -> {
+                JavacMethodElement(
+                    env = this,
+                    containing = enclosingType,
+                    element = element
+                )
+            }
+            else -> error("Unsupported kind ${element.kind} of executable element $element")
+        }
+    }
+
+    // TODO: Add a cache layer
+    fun wrapVariableElement(element: VariableElement): JavacVariableElement {
+        return when (val enclosingElement = element.enclosingElement) {
+            is ExecutableElement -> {
+                val executableElement = wrapExecutableElement(enclosingElement)
+
+                executableElement.parameters.find { param ->
+                    param.element == element
```

Comment:
java AP does not implement equals for these elements. In this case, i assume it will still work since it will be the same element instance, so we can also do === here. (== fine too)